storagenode/orders/service.go

```diff
@@ -71,102 +71,125 @@ type DB interface {
-	return &Sender{
+// NewService creates an order service.
+func NewService(log *zap.Logger, transport transport.Client, orders DB, trust *trust.Pool, config Config) *Service {
+	return &Service{
 		log:       log,
 		transport: transport,
 		orders:    orders,
 		config:    config,
 		trust:     trust,
 
-		Loop: *sync2.NewCycle(config.Interval),
+		Sender:  *sync2.NewCycle(config.SenderInterval),
+		Cleanup: *sync2.NewCycle(config.CleanupInterval),
 	}
 }
 
 // Run sends orders on every interval to the appropriate satellites.
-func (sender *Sender) Run(ctx context.Context) (err error) {
+func (service *Service) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
-	return sender.Loop.Run(ctx, sender.runOnce)
+
+	var group errgroup.Group
+	service.Sender.Start(ctx, &group, service.sendOrders)
+	service.Cleanup.Start(ctx, &group, service.cleanArchive)
+
+	return group.Wait()
+}
+
+func (service *Service) cleanArchive(ctx context.Context) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	service.log.Debug("cleaning")
+
+	deleted, err := service.orders.CleanArchive(ctx, service.config.ArchiveTTL)
+	if err != nil {
+		service.log.Error("cleaning archive", zap.Error(err))
+		return nil
+	}
+
+	service.log.Debug("cleanup finished", zap.Int("items deleted", deleted))
+	return nil
 }
 
-func (sender *Sender) runOnce(ctx context.Context) (err error) {
+func (service *Service) sendOrders(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
-	sender.log.Debug("sending")
+	service.log.Debug("sending")
 
 	const batchSize = 1000
 
-	ordersBySatellite, err := sender.orders.ListUnsentBySatellite(ctx)
+	ordersBySatellite, err := service.orders.ListUnsentBySatellite(ctx)
 	if err != nil {
 		if ordersBySatellite == nil {
-			sender.log.Error("listing orders", zap.Error(err))
+			service.log.Error("listing orders", zap.Error(err))
 			return nil
 		}
 
-		sender.log.Warn("DB contains invalid marshalled orders", zap.Error(err))
+		service.log.Warn("DB contains invalid marshalled orders", zap.Error(err))
 	}
 
 	requests := make(chan ArchiveRequest, batchSize)
 	var batchGroup errgroup.Group
-	batchGroup.Go(func() error { return sender.handleBatches(ctx, requests) })
+	batchGroup.Go(func() error { return service.handleBatches(ctx, requests) })
 
 	if len(ordersBySatellite) > 0 {
 		var group errgroup.Group
-		ctx, cancel := context.WithTimeout(ctx, sender.config.Timeout)
+		ctx, cancel := context.WithTimeout(ctx, service.config.SenderTimeout)
 		defer cancel()
 
 		for satelliteID, orders := range ordersBySatellite {
 			satelliteID, orders := satelliteID, orders
 			group.Go(func() error {
-				sender.Settle(ctx, satelliteID, orders, requests)
+				service.Settle(ctx, satelliteID, orders, requests)
 				return nil
 			})
 		}
 
 		_ = group.Wait() // doesn't return errors
 	} else {
-		sender.log.Debug("no orders to send")
+		service.log.Debug("no orders to send")
 	}
 
 	close(requests)
 	return batchGroup.Wait()
```

Comment:
Ensure that this func doesn't return an error and log it instead, otherwise the error will shutdown the node.